packages/devtools_app/lib/src/screens/inspector/inspector_tree_controller.dart

```diff
@@ -94,9 +94,9 @@ class _InspectorTreeRowState extends State<_InspectorTreeRowWidget>
     setState(() {
       final row = widget.row;
       if (expanded) {
-        widget.inspectorTreeState.controller.onExpandRow(row);
+        widget.inspectorTreeState.controller!.onExpandRow(row);
```

Comment:
make this controller also  non-nullable perhaps using late.

Reply:
This controller is sometimes null, so cannot be late. It is always non-null if the node is expanded.

Reply:
The controller can be null: https://github.com/flutter/devtools/blob/2a707ca56c1a9d5eeef212c28c573548a051fdd2/packages/devtools_app/lib/src/screens/inspector/inspector_tree_controller.dart#L964

Reply:
with more context of the other null checks in this widget, I agree with Jacob. Add a non-nullable parameter to the _InspectorTreeRowWidget for this controller. Then that can be used both in this method and in the build method above.